CredentialsManager/Files.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.IO;
+using System.Security;
+using CredentialsManager.FilesExceptions;
+
+namespace CredentialsManager
+{
+  public static class Files
+  {
+    private static readonly string[] Delimiters = {"\r\n", "\n"};
+    
+    public static string ReadAllText(string file)
+    {
+      try
+      {
+        using (var reader = new StreamReader(file))
+        {
+          return reader.ReadToEnd();
+        }
+      }
+      catch (Exception ex) when (ex is ArgumentException
+                              || ex is ArgumentNullException
+                              || ex is FileNotFoundException
+                              || ex is DirectoryNotFoundException
+                              || ex is IOException)
+      {
+        throw new MyFileNotFoundException($"{file} file not found ore not able to open!");
```

Comment:
You're kind of swallowing useful information here.  This whole catch / rethrow block feels a little iffy to me.  You could maybe just catch `FileNotFoundException` and translate it to provide a more helpful error message (although the `Message` set by `FileNotFoundException` is already pretty clear) but I wouldn't catch all of these.

Also, if you are going to catch and rethrow like this it is best to keep `ex` around as an "inner exception" (you can google this).  It keeps the stack trace of `ex` around instead of swallowing it.

Reply:
I decided to catch those exceptions because their summaries lead to one conclusion: the file, directory or path can not be found. The 'ReadToEnd()' method's summary for 'IOException' is indeed different from the one of 'StreamReader' object creation.
I refactored the constructor of `MyFileNotFoundException` to include the initial exception thus exposing the `InnerException` property. I hope I understood this correctly.
The commit for this update: [6ba6627](https://github.com/csinn/CSharp-From-Zero-To-Hero-v2/pull/43/commits/6ba6627217ee878b8d508018d4350ad78f7f5a5a)